data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/HibernateJpaOperations.java

```diff
@@ -334,12 +334,13 @@ public <T> T persist(@NonNull InsertOperation<T> operation) {
     public <T> T update(@NonNull UpdateOperation<T> operation) {
         StoredQuery<T, ?> storedQuery = operation.getStoredQuery();
         return transactionOperations.executeWrite(status -> {
+            EntityManager session = sessionFactory.getCurrentSession();
             if (storedQuery != null) {
                 executeEntityUpdate(storedQuery, operation.getEntity());
+                flushIfNecessary(session, operation.getAnnotationMetadata(), true);
```

Comment:
With the Hibernate Session we can remove just the entity submitted.

Reply:
Thanks, I missed there is entity present in several methods so I updated the code. Only left `session.clear()` in executeUpdate where we have no entities reference so must clear session, but only if user wants it via `QueryHint` and session auto flush.